src/LintableFile.js

```diff
@@ -0,0 +1,253 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import path from "node:path";
+import log4js from "log4js";
+import { getLocaleFromPath } from "ilib-tools-common";
+import { Fix, IntermediateRepresentation, Parser, Result, SourceFile, FileStats } from "ilib-lint-common";
+import DirItem from "./DirItem.js";
+import Project from "./Project.js";
+import FileType from "./FileType.js";
+
+const logger = log4js.getLogger("ilib-lint.root.LintableFile");
+
+/**
+ * @class Represent a source file
+ */
+class LintableFile extends DirItem {
+    /**
+     * Construct a source file instance
+     * The options parameter can contain any of the following properties:
+     *
+     *
+     * @constructor
+     * @param {String} filePath path to the source file
+     * @param {Object} options options for constructing this source file
+     * @param {FileType} options.filetype file type of this source file
+     * @param {String} options.filePath path to the file
+     * @param {object} [options.settings] additional settings from the ilib-lint config that apply to this file
+     * @param {Project} project the project where this file is located
+     */
+    constructor(filePath, options, project) {
+        super(filePath, options, project);
+        if (!options || !filePath || !options.filetype) {
+            throw "Incorrect options given to LintableFile constructor";
+        }
+        this.sourceFile = new SourceFile(filePath, {
+            sourceLocale: options.locale,
+            getLogger: log4js.getLogger.bind(log4js),
+            type: options.filetype.getName()
+        });
+        this.filetype = options.filetype;
+
+        if (this.project.options.opt && this.project.options.opt.verbose) {
+            logger.level = "debug";
+        }
+        this.parsers = [];
+        let extension = path.extname(filePath);
+        if (extension) {
+            // remove the dot
+            extension = extension.substring(1);
+            this.parsers = this.filetype.getParserClasses(extension);
+        }
+    }
+
+    /**
+     * Return the locale gleaned from the file path using the template in
+     * the settings, or undefined if no locale could be found.
+     *
+     * @returns {String} the locale gleaned from the path, or the empty
+     * string if no locale could be found.
+     */
+    getLocaleFromPath() {
+        if (this.settings && this.settings.template) {
+            return getLocaleFromPath(this.settings.template, this.sourceFile.getPath());
+        }
+        return "";
```

Comment:
Is there a particular reason to return an empty string here? I feel like `undefined` or `null` better indicate that the information was not found

Reply:
This depends on getLocaleFromPath from `ilib-lint-common` which also returns an empty string if the locale is not found. For now, since this code was not modified in this PR, I would like to not change it.